src/Admin/AbstractAdmin.php

```diff
@@ -3123,6 +3123,29 @@ private function buildRoutes(): void
             $extension->configureRoutes($this, $this->routes);
         }
     }
+
+    /**
+     * Determines default sorting of list (name of field and direction).
+     * Returns an array with 2 keys: "_sort_by" - name of field, "_sort_order" - sorting direction.
+     *
+     * Looks for sorting parameters in this order (first found will be used):
+     * 1. $this->$datagridValues property
+     * 2. Query used to load data
+     * 3. Defaults based on entity
+     *
+     * @return array
+     */
+    private function determineSorting()
```

Comment:
Please, add return type declaration and remove docblock.

Reply:
Done in 6ea8b2a1 commit

Reply:
IMO, the description was right, since you was explaining the logic and the structure this method returns.
BTW, could you please fix the tests and cover these changes?

Reply:
IMHO, it will be nice to explain how it works. Should I restore the PhpDoc (revert 6ea8b2a commit)?

Reply:
Yes, but add just the description. The `@return array` docblock wasn't adding any value.

Reply:
Here you go: f2bd4b39 commit 🙃

Reply:
Here you go: f2bd4b39 commit 🙃